components/processor/gcomapiprocessor/internal/gcom/client/client.go

```diff
@@ -0,0 +1,154 @@
+package client
+
+import (
+	"context"
+	"flag"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/opentelemetry-collector-components/processor/gcomapiprocessor/internal/gcom/common"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	grafanaComReqs = promauto.NewHistogramVec(
```

Comment:
Do we want to use the telemetry settings that the collector provides? While the registries might end up merged, this is likely to change in the future.